modules/effects/src/actions.ts

```diff
@@ -87,12 +108,27 @@ export function ofType<
  * arguments, to preserve backwards compatibility with old versions of ngrx.
  */
 export function ofType<V extends Action>(
-  ...allowedTypes: string[]
+  ...allowedTypes: Array<string | ActionCreator<string, Creator>>
 ): OperatorFunction<Action, V>;
 export function ofType(
-  ...allowedTypes: string[]
+  ...allowedTypes: Array<string | ActionCreator<string, Creator>>
 ): OperatorFunction<Action, Action> {
-  return filter((action: Action) =>
-    allowedTypes.some(type => type === action.type)
-  );
+  return filter((action: Action) => {
+    for (let typeOrActionCreator of allowedTypes) {
```

Comment:
What about checking on action first and next on the action creator type?

```ts
return allowedTypes.some(type => {
  // Comparing the string to type
  if (typeof type === 'string') {
    return type === action.type;
  }

  // We are filtering by ActionCreator
  return type && type.type === action.type;
});
```

Reply:
You are right, reads much better. Also I think I reimplemented `some` in my previous solution 🤦‍♂️